packages/web/src/store/selectors/scanDisplayOptions/scanDisplayOptions.ts

```diff
@@ -0,0 +1,59 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { RootState } from '../../index';
+import { ScanDisplayOptions } from '../../slices/scanDisplayOptions';
+import { makeSelectScanPackagesByUrl } from '../websiteResults';
+import { packageFilterPredicates } from './packageFilters';
+import { sortPackagesByOptions } from './packageSorters';
+
+export const DEFAULT_SCAN_DISPLAY_OPTIONS: ScanDisplayOptions = {
+  packageFilters: {
+    keywords: [],
+    authors: [],
+    traits: [],
+  },
+  packageSorters: [
+    {
+      by: 'severity',
+      direction: 'DESC',
+    },
+    {
+      by: 'popularity',
+      direction: 'DESC',
+    },
+  ],
+};
+
+export const makeSelectScanDisplayOptions = () =>
+  createSelector(
+    [(state: RootState) => state.scanDisplayOptions, (_, scanUrl: string | undefined) => scanUrl],
+    (scanDisplayOptions, scanUrl = '') =>
+      scanDisplayOptions[scanUrl] ?? DEFAULT_SCAN_DISPLAY_OPTIONS
+  );
+
+export const makeSelectOptimizedScanDisplayOptions = () =>
```

Comment:
Why do you need to have both optimized and non-optimized scan display options? Can't we use sets everywhere?

Reply:
That's in the plans for refactoring pass.